handle_actions.go

```diff
@@ -4,25 +4,24 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/labstack/echo"
+	"github.com/labstack/echo/v4"
 	"github.com/parnurzeal/gorequest"
 	"github.com/sirupsen/logrus"
 )
 
-func (ms MocksArray) DoActions(ctx Context) error {
+func (ms MocksArray) DoActions(ctx Context) {
```

Comment:
did a refactor here so that DoActions will stop if it finds the first match constraint. This change also implies that when creating yaml files, put the stronger constraints mocks before others.